eox_core/pipeline.py

```diff
@@ -164,3 +171,24 @@ def assert_user_information(details, user, backend, *args, **kwargs):
                 **locals()
             )
             raise AuthFailed(backend, "Credentials not allowed.")  # pylint: disable=raising-non-exception
+
+
+def ensure_user_has_signup_source(user=None, *args, **kwargs):
+    """
+    This pipeline step registers a new signup source for the current user and site.
+
+    It's recommended to place this step at the end of the TPA pipeline, e.g., after the step
+    'eox_core.pipeline.ensure_user_has_profile',
+
+    To avoid errors, these steps avoid signup source creation during the registration process.
+    """
+    if not user or kwargs.get("is_new"):
```

Comment:
Could you please explain in the docstring or a comment where is this is_new comming from?

I don't really care about skipping this for is_new. I am however making the push to have 2 steps. One that filters by ["new_association"](https://github.com/python-social-auth/social-core/blob/master/social_core/pipeline/social_auth.py#L54) and one that does not.

Reply:
Also on the topic of `kwargs.get("is_new")`. Should someone put this step before [social_user](https://github.com/python-social-auth/social-core/blob/master/social_core/pipeline/social_auth.py#L28) we would have a key_error rendering a very confusing 500 error to the final user.

Would you consider using `kwargs.get("is_new", False)`?

Reply:
Thanks for your suggestions, I updated the PR taking them into account.